### PR TITLE
Add ability to cancel DLC

### PR DESCRIPTION
**Pull request by benthecarman**
Adds the ability to cancel a DLC, deleting all the data and unreserving the utxos.

This is done in the GUI by right clicking in the table view and selecting `Cancel DLC`

### CODE REVIEW

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala [hunk 1]
```diff
@@ -339,6 +340,41 @@ class DLCPaneModel(resultArea: TextArea, oracleInfoArea: TextArea)
                               this)
   }
 
+  def cancelDLC(status: DLCStatus): Unit = {
+    if (status.state == DLCState.Offered || status.state == DLCState.Accepted) {
```
Comment (Christewart) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala]: Please pattern match on the states so we have exhaustiveness warnings if we add a new `DLCState` in the future.

**Review by Christewart — CHANGES_REQUESTED** [review]

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala [hunk 2]
```diff
@@ -339,6 +340,43 @@ class DLCPaneModel(resultArea: TextArea, oracleInfoArea: TextArea)
                               this)
   }
 
+  def cancelDLC(status: DLCStatus): Unit = {
+    status.state match {
+      case DLCState.Offered | DLCState.Accepted =>
+        val confirmed = new Alert(AlertType.Confirmation) {
+          initOwner(owner)
+          headerText = "Confirm Canceling DLC"
+          contentText = "Are you sure you want to cancel this DLC?\n" +
```
Comment (nkohen) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala]: should probably, at the very least, include the `event_id` if not more information from the `status` in this confirm alert

Comment (benthecarman) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala]: added

File: dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/WalletDLCSetupTest.scala [hunk 3]
```diff
@@ -355,6 +356,70 @@ class WalletDLCSetupTest extends BitcoinSDualWalletTest {
...
+        balance <- walletB.getBalance()
+        reserved <- walletB.spendingInfoDAO.findByTxoState(TxoState.Reserved)
+        dlcOpt <- walletB.findDLC(offer.paramHash)
+      } yield {
+        assert(balance == oldBalance)
+        assert(reserved.isEmpty)
+        assert(dlcOpt.isEmpty)
+      }
+  }
+
```
Comment (Christewart) [inline dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/WalletDLCSetupTest.scala:422]: Would be nice to have a negative test case here for failing to cancel a dlc that is not in a cancellable state

**Review by Christewart — APPROVED** [review]

**Pull request merged**